include/envoy/stats/stats.h

```diff
@@ -371,6 +388,8 @@ class StoreRoot : public Store {
    * method would be asserted.
    */
   virtual void mergeHistograms(PostMergeCb merge_complete_cb) PURE;
+
+  virtual FlushDelegate& flushDelegate() PURE;
```

Comment:
nit: Could StoreRoot just implement FlushSource? Any reason to have an accessor?

Reply:
Yeah, I went back and forth on this.  I did it this way for 2 reasons:
1.  Modularity: `FlushSource` is basically a cache that could be implemented as a totally separate object that just uses a store's public interface as a metric source.  I didn't want to necessarily impose structure that didn't need to be there, and if we ever want to make a StoreRoot implement it, we have the flexibility to make that decision on a per-implementation basis and just return `*this`.  Also, the cache implementation is simple enough that it's nice to be able to have a single impl for wherever it might be needed.
2.  This is more aesthetic, but I thought it was a little confusing to have the `cachedCounters()` and `counters()`, etc methods sitting next to each other in the same object (although that would happen anyway if we ever decide to make `ThreadLocalStore` implement `FlushSource`).

Anyway, I could go either way on this, so I'm happy to change if the other way seems better.

Reply:
I don't feel strongly about it. I guess it does relate to my question below as to do we even need the caching stuff right now?